src/Pos/GState/Context.hs

```diff
@@ -13,46 +14,58 @@ import           Universum
 
 import           Control.Lens           (makeClassy)
 import           System.Wlog            (WithLogger)
+import           Ether.Internal         (HasLens (..))
 
 import           Pos.Block.Slog.Context (SlogContext, cloneSlogContext)
 import           Pos.DB.Pure            (DBPureVar, cloneDBPure)
+import           Pos.DB.Rocks           (NodeDBs)
 import           Pos.Lrc.Context        (LrcContext, cloneLrcContext)
 import           Pos.Slotting           (SlottingData, cloneSlottingVar)
 
+data DBSum = RealDB NodeDBs | PureDB DBPureVar
+
 -- | This type contains DB and in-memory contexts which basically
 -- replicate GState. It's parameterized by DB type, because we have
 -- multiple DB implementations.
 --
 -- [CSL-1390] FIXME: add SSC GState here too!
-data GStateContext db = GStateContext
-    { _gscDB          :: !db
-    , _gscLrcContext  :: !LrcContext
-    , _gscSlogContext :: !SlogContext
-    , _gscSlottingVar :: !(TVar SlottingData)
+data GStateContext = GStateContext
+    { _gscDB          :: !DBSum
+    , _gscLrcContext  :: LrcContext
+    , _gscSlogContext :: SlogContext
+    , _gscSlottingVar :: (TVar SlottingData)
```

Comment:
Why did you make these types lazy? If it's intentional, you must write a comment, because by default we make everything strict and someone can change it accidentally.